hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -345,16 +358,59 @@ public void doMetadataTableValidation() {
     boolean finalResult = true;
     metaClient.reloadActiveTimeline();
     String basePath = metaClient.getBasePath();
+    List<String> baseFilesUnderDeletion = Collections.emptyList();
+
+    if (cfg.skipUnderDeletionDataFiles) {
+      HoodieTimeline pendingCleaningTimeline = metaClient.getActiveTimeline()
+          .getCleanerTimeline()
+          .filter(instant -> instant.getState() != HoodieInstant.State.COMPLETED);
+
+      baseFilesUnderDeletion = pendingCleaningTimeline.getInstants().flatMap(instant -> {
+        try {
+          if (instant.isInflight()) {
+            // convert inflight instant to requested and get clean plan
+            instant = new HoodieInstant(HoodieInstant.State.REQUESTED, instant.getAction(), instant.getTimestamp());
+          }
+          HoodieCleanerPlan cleanerPlan = CleanerUtils.getCleanerPlan(metaClient, instant);
+
+          return cleanerPlan.getFilePathsToBeDeletedPerPartition().values().stream().flatMap(cleanerFIleInfoList -> {
+            return cleanerFIleInfoList.stream().map(fileInfo -> {
+              return new Path(fileInfo.getFilePath()).getName();
+            });
+          });
+
+        } catch (HoodieIOException ex) {
+
+          if (ex.getIOException() instanceof FileNotFoundException) {
+            // cleaner instant could be deleted by archive and FileNotFoundException could be threw during getInstantDetails function
```

Comment:
Inflight cleaning instants shouldn't be archived.  Is this considering the inflight cleaning instant deleted due to failed cleaning?

Reply:
Actually no need to check FileNotFoundException here. removed.